runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkProcessContext.java

```diff
@@ -127,34 +127,10 @@ public PipelineOptions getPipelineOptions() {
   }
 
   @Override
-  public void output(OutputT output) {
-    outputWithTimestamp(output, windowedValue != null ? windowedValue.getTimestamp() : null);
-  }
-
-  public abstract void output(WindowedValue<OutputT> output);
-
-  @Override
-  public <T> void sideOutput(TupleTag<T> tupleTag, T t) {
-    String message = "sideOutput is an unsupported operation for doFunctions, use a "
-        + "MultiDoFunction instead.";
-    LOG.warn(message);
-    throw new UnsupportedOperationException(message);
-  }
-
-  @Override
-  public <T> void sideOutputWithTimestamp(TupleTag<T> tupleTag, T t, Instant instant) {
```

Comment:
The base class delegates normal `output()` and `sideOutput()` on the `ProcessContext` to the internal abstract methods I introduced - `outputWindowedValue` and `sideOutputWindowedValue` (I renamed them and made them protected in response to your comments). `sideOutputWindowedValue` in `DoFnFunction` is overridden to throw an `UnsupportedOperationException`.